addons/hw_drivers/iot_handlers/drivers/PrinterDriver.py

```diff
@@ -208,7 +209,94 @@ def format_escpos(self, im):
+        width_pixels, height_pixels = im.size
+        for left in range(0, width_pixels, line_height):
+            box = (left, 0, left + line_height, height_pixels)
+            im_chunk = im.transform(
+                (line_height, height_pixels),
+                Image.EXTENT,
+                box
+            )
+            yield im_chunk.tobytes()
+
+    def format_escpos_bit_image_column(self, im, high_density_vertical=True,
+                                       high_density_horizontal=True,
+                                       size_scale=100):
+        """ prints with the `ESC *`-command
+        reference: https://reference.epson-biz.com/modules/ref_escpos/index.php?content_id=88
+
+        :param im: PIL image to print
+        :param high_density_vertical: print in high density in vertical direction
+        :param high_density_horizontal: print in high density in horizontal direction
+        :param size_scale: picture scale in percentage,
+        e.g: 50 -> half the size (horizontally and vertically)
+        """
+        size_scale_ratio = size_scale / 100
+        size_scale_width = int(im.width * size_scale_ratio)
+        size_scale_height = int(im.height * size_scale_ratio)
+        im = im.resize((size_scale_width, size_scale_height))
+        # escpos ESC * command print column per column
+        # (instead of usual row by row).
+        # So we transpose the picture to ease the calculations
+        im = im.transpose(Image.ROTATE_270).transpose(Image.FLIP_LEFT_RIGHT)
+
+        ESC = b'\x1b'
+        density_byte = (1 if high_density_horizontal else 0) + \
+                       (32 if high_density_vertical else 0)
+        nL = im.height & 0xFF
+        nH = (im.height >> 8) & 0xFF
+        HEADER = ESC + b'*' + bytes([density_byte, nL, nH])
+
+        raster_data = ESC + b'3\x10'  # Adjust line-feed size
+        line_height = 24 if high_density_vertical else 8
+        for column in self.extract_columns_from_picture(im, line_height):
+            raster_data += HEADER + column + b'\n'
+        raster_data += ESC + b'2'  # Reset line-feed size
+        return raster_data
+
+    def format_escpos(self, im):
+        # Epson support different command to print pictures.
+        # We use by default "GS v 0", but it  is incompatible with certain
+        # printer models (like TM-U2x0)
+        # As we are pretty limited in the information that we have, we will
+        # use the printer name to parse some configuration value
+
+        # Default values
+        image_mode = 'raster'
+        high_density_vertical = True
+        high_density_horizontal = True
+        scale = 100
+
+        # Parse the printer name to get the needed parameters
+        # The separator need to not be filtered by `get_identifier`
+        options = self.device_name.split('_')
+        for option in options:
+            option = option.upper()
+            if option == 'IMC':
```

Comment:
Can we think of more "complex"-looking name for these configuration strings? The current choices feel like they can be easily part of existing (default) printer names.

Reply:
Proposed: `<PrinterName>__IMC_LDV_LDH_SCALE70__`

The idea is to isolate the options from the name of the printer.

`<printername>__<options>__`